editor/components/rich-text/index.js

```diff
@@ -592,9 +589,10 @@ export class RichText extends Component {
 	 * before the selection. Sends the elements after the selection to the `onSplit`
 	 * handler.
 	 *
-	 * @param {Array} blocks The blocks to add after the split point.
+	 * @param {Array}  blocks  The blocks to add after the split point.
+	 * @param {Object} context The context for splitting.
 	 */
-	splitContent( blocks = [] ) {
+	splitContent( blocks = [], context = {} ) {
```

Comment:
It seems paste context is only used to pass null instead of empty values in before/after. If that's case maybe we can consider doing the empty checks on the onSplit e.g if( isEmpty( before ) ) instead of if( before ) to avoid adding more complexity in the RichText.

Reply:
You mean in the block registration file? The problem is that it's `RichText` specific, unless we maybe add it as `RichText.isEmpty( value )`. What do you think?

Reply:
I was not aware it was RichText specific I thought a simple lodash isEmpty would be ok. In this case, I think we can leave the solution we have now :)